aea/crypto/ethereum.py

```diff
@@ -102,26 +104,28 @@ def _load_private_key_from_path(self, file_name) -> Account:
         except IOError as e:  # pragma: no cover
             logger.exception(str(e))
 
-    def sign_transaction(self, tx_hash: bytes) -> bytes:
+    def sign_message(self, tx_hash: bytes) -> bytes:
         """
         Sign a transaction hash.
 
         :param tx_hash: the transaction hash
         :return: Signed message in bytes
         """
-        signature = self.entity.signHash(tx_hash)
+        tx = encode_defunct(primitive=tx_hash)
+        signature = self.entity.sign_message(tx)
         return signature["signature"]
 
-    # def recover_from_hash(self, tx_hash: bytes, signature: bytes) -> Address:
-    #     """
-    #     Recover the address from the hash.
+    def recover_from_hash(self, tx_hash: bytes, signature: bytes) -> Address:
```

Comment:
let's call it `recover_message(self, message: bytes, signature: bytes) -> Address`